xmipy/xmiwrapper.py

```diff
@@ -80,10 +82,15 @@ def __init__(
     @staticmethod
     def _add_lib_dependency(lib_dependency: Union[str, Path]) -> None:
         lib_dependency = str(lib_dependency)
-        if platform.system() == "Windows":
-            os.environ["PATH"] = lib_dependency + os.pathsep + os.environ["PATH"]
-        else:
-            # Assume a Unix-like system
+        sysinfo = platform.system()
+        if sysinfo == "Windows":
+            try:
+                os.add_dll_directory(lib_dependency)  # type: ignore
```

Comment:
I've tried it again with MetaSWAP, and calling `add_dll_directory` is not enough to let it find the dll dependencies.
@mwtoews if you are interested in looking into this and don't have a suitable library available you can write to julian.hofer@deltares.nl and I will provide you with the MetaSWAP dll to experiment.

Otherwise, it would be nice if you could revert this part of the PR and I only merge the DYLD_FALLBACK_LIBRARY_PATH part.